### PR TITLE
[CORTX-DEV] EOS-14335-cortx-dev: Alerts not visible for Pvt data nw

### PR DESCRIPTION
**Pull request by pawankumarsrivastava**
# Backend

## Problem Statement
<pre>
Story Ref (if any): https://jts.seagate.com/browse/EOS-14335
Private data network port down on a node where CSM is running is not seen by CSM
</pre>
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
When Pvt nw is down elasticsearch also went down. So, when CSM receives the alert due to elasticsearch is still in cloning process CSM is not able to save the alert in db. Also, CSM ack that alert and it omits from RMQ. 
</pre>
## Solution
<pre>
Elasticsearch cloning requires few seconds so we will try to first connect to elasticsearch and then start the process of saving the alerts. Also, in case of any exception we will not ack the alert so that it remains in RMQ.
</pre>
## Unit Test Cases
<pre>

</pre>
Signed-off-by: pawan.kumarsrivastava@seagate.com

### CODE REVIEW

File: csm/core/services/alerts.py [hunk 1]
```diff
@@ -609,6 +609,27 @@ def stop(self):
...
+        """
+        prev_alert = None
+        try:
+            prev_alert = self._run_coroutine\
+                (self.repo.retrieve_by_sensor_info(sensor_info, module_type))
+            return prev_alert
+        except Exception as ex:
+            Log.warn(f"Unabke to fetch previous alert.{ex}")
+            time.sleep(5)
+            self._get_previous_alert(sensor_info, module_type)
```
Comment (ajaysrivas) [inline csm/core/services/alerts.py:631]: Instead limit the calls to some configurable retries.

File: csm/plugins/cortx/alert.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def _plugin_callback(self, message):
                     """
                     Calling HA Decision Maker for Alerts.
                     """
-                    if self.decision_maker_service:
+                    if self.decision_maker_service and status:
                         self.decision_maker_service.decision_maker_callback(sensor_queue_msg)
             except ValidationError as ve:
```
Comment (ajaysrivas) [inline csm/plugins/cortx/alert.py:181]: Acknowledge in ValidationError but not in other exceptions.

**Review by ajaysrivas — CHANGES_REQUESTED** [review]

File: csm/core/services/alerts.py (hunk 1 above)
Comment (pawankumarsrivastava) [reply csm/core/services/alerts.py]: Done

File: csm/plugins/cortx/alert.py (hunk 2 above)
Comment (pawankumarsrivastava) [reply csm/plugins/cortx/alert.py:181]: Done

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**